File: power-square-android/app/src/main/java/com/okla/ops/beans/SettingInfoBean.java
package com.okla.ops.beans;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

import com.okla.ops.BR;

/**
 * @Date: 2021/3/5 11:04
 * @Author: Craz
 * @Description:
 * @Version:
 */
public class SettingInfoBean extends BaseObservable {

    /**
     * emptySpaceDetection : {"count":9}
     * noCharging : {"blackoutTime":"14:40","chargingTime":"14:43"}
     * restartTheChargerRegularly : {"open":1,"time":"11:8"}
     * heating : {"pmsAutoControl":0,"apkControl":{"open":0,"permit":15,"prohibit":28}}
     * homePageDetails : {"show":1}
     * microSwitch : {"open":1}
     * batteryConnector : {"open":1}
     * chargingStrategy : {"type":1,"level":null}
     * minimumElectricityBorrowed : {"level":70}
     * maximumChargeCapacity : {"level":100}
     */

    private EmptySpaceDetectionBean emptySpaceDetection;
    private NoChargingBean noCharging;
    private RestartTheChargerRegularlyBean restartTheChargerRegularly;
    private HeatingBean heating;
    private HomePageDetailsBean homePageDetails;
    private MicroSwitchBean microSwitch;
    private BatteryConnectorBean batteryConnector;
    private ChargingStrategyBean chargingStrategy;
    private MinimumElectricityBorrowedBean minimumElectricityBorrowed;
    private MaximumChargeCapacityBean maximumChargeCapacity;

    @Bindable
    public EmptySpaceDetectionBean getEmptySpaceDetection() {
        return emptySpaceDetection;
    }

    public void setEmptySpaceDetection(EmptySpaceDetectionBean emptySpaceDetection) {
        this.emptySpaceDetection = emptySpaceDetection;
        notifyPropertyChanged(BR.emptySpaceDetection);
    }

    @Bindable
    public NoChargingBean getNoCharging() {
        return noCharging;
    }

    public void setNoCharging(NoChargingBean noCharging) {
        this.noCharging = noCharging;
        notifyPropertyChanged(BR.noCharging);
    }

    @Bindable
    public RestartTheChargerRegularlyBean getRestartTheChargerRegularly() {
        return restartTheChargerRegularly;
    }

    public void setRestartTheChargerRegularly(RestartTheChargerRegularlyBean restartTheChargerRegularly) {
        this.restartTheChargerRegularly = restartTheChargerRegularly;
        notifyPropertyChanged(BR.restartTheChargerRegularly);
    }

    @Bindable
    public HeatingBean getHeating() {
        return heating;
    }

    public void setHeating(HeatingBean heating) {
        this.heating = heating;
        notifyPropertyChanged(BR.heating);
    }

    @Bindable
    public HomePageDetailsBean getHomePageDetails() {
        return homePageDetails;
    }

    public void setHomePageDetails(HomePageDetailsBean homePageDetails) {
        this.homePageDetails = homePageDetails;
        notifyPropertyChanged(BR.homePageDetails);
    }

    @Bindable
    public MicroSwitchBean getMicroSwitch() {
        return microSwitch;
    }

    public void setMicroSwitch(MicroSwitchBean microSwitch) {
        this.microSwitch = microSwitch;
        notifyPropertyChanged(BR.microSwitch);
    }

    @Bindable
    public BatteryConnectorBean getBatteryConnector() {
        return batteryConnector;
    }

    public void setBatteryConnector(BatteryConnectorBean batteryConnector) {
        this.batteryConnector = batteryConnector;
        notifyPropertyChanged(BR.batteryConnector);
    }

    @Bindable
    public ChargingStrategyBean getChargingStrategy() {
        return chargingStrategy;
    }

    public void setChargingStrategy(ChargingStrategyBean chargingStrategy) {
        this.chargingStrategy = chargingStrategy;
        notifyPropertyChanged(BR.chargingStrategy);
    }

    @Bindable
    public MinimumElectricityBorrowedBean getMinimumElectricityBorrowed() {
        return minimumElectricityBorrowed;
    }

    public void setMinimumElectricityBorrowed(MinimumElectricityBorrowedBean minimumElectricityBorrowed) {
        this.minimumElectricityBorrowed = minimumElectricityBorrowed;
        notifyPropertyChanged(BR.minimumElectricityBorrowed);
    }

    @Bindable
    public MaximumChargeCapacityBean getMaximumChargeCapacity() {
        return maximumChargeCapacity;
    }

    public void setMaximumChargeCapacity(MaximumChargeCapacityBean maximumChargeCapacity) {
        this.maximumChargeCapacity = maximumChargeCapacity;
        notifyPropertyChanged(BR.maximumChargeCapacity);
    }

    public static class EmptySpaceDetectionBean extends BaseObservable {
        /**
         * count : 9
         */

        private int count;

        @Bindable
        public int getCount() {
            return count;
        }

        public void setCount(int count) {
            this.count = count;
            notifyPropertyChanged(BR.count);
        }
    }

    public static class NoChargingBean extends BaseObservable {
        /**
         * blackoutTime : 14:40
         * chargingTime : 14:43
         */

        private String blackoutTime;
        private String chargingTime;

        @Bindable
        public String getBlackoutTime() {
            return blackoutTime;
        }

        public void setBlackoutTime(String blackoutTime) {
            this.blackoutTime = blackoutTime;
            notifyPropertyChanged(BR.blackoutTime);
        }

        @Bindable
        public String getChargingTime() {
            return chargingTime;
        }

        public void setChargingTime(String chargingTime) {
            this.chargingTime = chargingTime;
            notifyPropertyChanged(BR.chargingTime);
        }
    }

    public static class RestartTheChargerRegularlyBean extends BaseObservable {
        /**
         * open : 1 ,1-开启 0-关闭
         * time : 11:8
         */

        private int open;
        private String time;

        @Bindable
        public int getOpen() {
            return open;
        }

        public void setOpen(int open) {
            this.open = open;
            notifyPropertyChanged(BR.open);
        }

        @Bindable
        public String getTime() {
            return time;
        }

        public void setTime(String time) {
            this.time = time;
            notifyPropertyChanged(BR.time);
        }
    }

    public static class HeatingBean extends BaseObservable {
        /**
         * pmsAutoControl : 0  ,0-关闭，1-开启
         * apkControl : {"open":0,"permit":15,"prohibit":28}
         */

        private int pmsAutoControl;
        private ApkControlBean apkControl;

        @Bindable
        public int getPmsAutoControl() {
            return pmsAutoControl;
        }

        public void setPmsAutoControl(int pmsAutoControl) {
            this.pmsAutoControl = pmsAutoControl;
            notifyPropertyChanged(BR.pmsAutoControl);
        }

        @Bindable
        public ApkControlBean getApkControl() {
            return apkControl;
        }

        public void setApkControl(ApkControlBean apkControl) {
            this.apkControl = apkControl;
            notifyPropertyChanged(BR.apkControl);
        }

        public static class ApkControlBean extends BaseObservable {
            /**
             * open : 0
             * permit : 15
             * prohibit : 28
             */

            private int open;
            private int permit;
            private int prohibit;

            @Bindable
            public int getOpen() {
                return open;
            }

            public void setOpen(int open) {
                this.open = open;
                notifyPropertyChanged(BR.open);
            }

            @Bindable
            public int getPermit() {
                return permit;
            }

            public void setPermit(int permit) {
                this.permit = permit;
                notifyPropertyChanged(BR.permit);
            }

            @Bindable
            public int getProhibit() {
                return prohibit;
            }

            public void setProhibit(int prohibit) {
                this.prohibit = prohibit;
                notifyPropertyChanged(BR.prohibit);
            }
        }
    }

    public static class HomePageDetailsBean extends BaseObservable {
        /**
         * show : 1
         */

        private int show;

        @Bindable
        public int getShow() {
            return show;
        }

        public void setShow(int show) {
            this.show = show;
            notifyPropertyChanged(BR.show);
        }
    }

    public static class MicroSwitchBean extends BaseObservable {
        /**
         * open : 1
         */

        private int open;

        @Bindable
        public int getOpen() {
            return open;
        }

        public void setOpen(int open) {
            this.open = open;
            notifyPropertyChanged(BR.open);
        }
    }

    public static class BatteryConnectorBean extends BaseObservable {
        /**
         * open : 1
         */

        private int open;

        @Bindable
        public int getOpen() {
            return open;
        }

        public void setOpen(int open) {
            this.open = open;
            notifyPropertyChanged(BR.open);
        }
    }

    public static class ChargingStrategyBean extends BaseObservable {
        /**
         * type : 1
         * level : null
         */

        private int type;
        private int level;

        @Bindable
        public int getType() {
            return type;
        }

        public void setType(int type) {
            this.type = type;
            notifyPropertyChanged(BR.type);
        }

        @Bindable
        public int getLevel() {
            return level;
        }

        public void setLevel(int level) {
            this.level = level;
            notifyPropertyChanged(BR.level);
        }
    }

    public static class MinimumElectricityBorrowedBean extends BaseObservable {
        /**
         * level : 70
         */

        private int level;

        @Bindable
        public int getLevel() {
            return level;
        }

        public void setLevel(int level) {
            this.level = level;
            notifyPropertyChanged(BR.level);
        }
    }

    public static class MaximumChargeCapacityBean extends BaseObservable {
        /**
         * level : 100
         */

        private int level;

        @Bindable
        public int getLevel() {
            return level;
        }

        public void setLevel(int level) {
            this.level = level;
            notifyPropertyChanged(BR.level);
        }
    }
}

File: lib/data/models/setting_info_bean.dart
import 'package:flutter/foundation.dart';

@immutable
class SettingInfoBean {
  final EmptySpaceDetectionBean? emptySpaceDetection;
  final NoChargingBean? noCharging;
  final RestartTheChargerRegularlyBean? restartTheChargerRegularly;
  final HeatingBean? heating;
  final HomePageDetailsBean? homePageDetails;
  final MicroSwitchBean? microSwitch;
  final BatteryConnectorBean? batteryConnector;
  final ChargingStrategyBean? chargingStrategy;
  final MinimumElectricityBorrowedBean? minimumElectricityBorrowed;
  final MaximumChargeCapacityBean? maximumChargeCapacity;

  const SettingInfoBean({
    required this.emptySpaceDetection,
    required this.noCharging,
    required this.restartTheChargerRegularly,
    required this.heating,
    required this.homePageDetails,
    required this.microSwitch,
    required this.batteryConnector,
    required this.chargingStrategy,
    required this.minimumElectricityBorrowed,
    required this.maximumChargeCapacity,
  });

  factory SettingInfoBean.fromJson(Map<String, dynamic> json) {
    return SettingInfoBean(
      emptySpaceDetection: json['emptySpaceDetection'] is Map<String, dynamic>
          ? EmptySpaceDetectionBean.fromJson(
              Map<String, dynamic>.from(json['emptySpaceDetection'] as Map),
            )
          : null,
      noCharging: json['noCharging'] is Map<String, dynamic>
          ? NoChargingBean.fromJson(
              Map<String, dynamic>.from(json['noCharging'] as Map),
            )
          : null,
      restartTheChargerRegularly:
          json['restartTheChargerRegularly'] is Map<String, dynamic>
              ? RestartTheChargerRegularlyBean.fromJson(
                  Map<String, dynamic>.from(
                    json['restartTheChargerRegularly'] as Map,
                  ),
                )
              : null,
      heating: json['heating'] is Map<String, dynamic>
          ? HeatingBean.fromJson(
              Map<String, dynamic>.from(json['heating'] as Map),
            )
          : null,
      homePageDetails: json['homePageDetails'] is Map<String, dynamic>
          ? HomePageDetailsBean.fromJson(
              Map<String, dynamic>.from(json['homePageDetails'] as Map),
            )
          : null,
      microSwitch: json['microSwitch'] is Map<String, dynamic>
          ? MicroSwitchBean.fromJson(
              Map<String, dynamic>.from(json['microSwitch'] as Map),
            )
          : null,
      batteryConnector: json['batteryConnector'] is Map<String, dynamic>
          ? BatteryConnectorBean.fromJson(
              Map<String, dynamic>.from(json['batteryConnector'] as Map),
            )
          : null,
      chargingStrategy: json['chargingStrategy'] is Map<String, dynamic>
          ? ChargingStrategyBean.fromJson(
              Map<String, dynamic>.from(json['chargingStrategy'] as Map),
            )
          : null,
      minimumElectricityBorrowed:
          json['minimumElectricityBorrowed'] is Map<String, dynamic>
              ? MinimumElectricityBorrowedBean.fromJson(
                  Map<String, dynamic>.from(
                    json['minimumElectricityBorrowed'] as Map,
                  ),
                )
              : null,
      maximumChargeCapacity: json['maximumChargeCapacity'] is Map<String,
              dynamic>
          ? MaximumChargeCapacityBean.fromJson(
              Map<String, dynamic>.from(json['maximumChargeCapacity'] as Map),
            )
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'emptySpaceDetection': emptySpaceDetection?.toJson(),
        'noCharging': noCharging?.toJson(),
        'restartTheChargerRegularly': restartTheChargerRegularly?.toJson(),
        'heating': heating?.toJson(),
        'homePageDetails': homePageDetails?.toJson(),
        'microSwitch': microSwitch?.toJson(),
        'batteryConnector': batteryConnector?.toJson(),
        'chargingStrategy': chargingStrategy?.toJson(),
        'minimumElectricityBorrowed': minimumElectricityBorrowed?.toJson(),
        'maximumChargeCapacity': maximumChargeCapacity?.toJson(),
      };
}

@immutable
class EmptySpaceDetectionBean {
  final int count;

  const EmptySpaceDetectionBean({
    required this.count,
  });

  factory EmptySpaceDetectionBean.fromJson(Map<String, dynamic> json) {
    return EmptySpaceDetectionBean(
      count: (json['count'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'count': count,
      };
}

@immutable
class NoChargingBean {
  final String? blackoutTime;
  final String? chargingTime;

  const NoChargingBean({
    required this.blackoutTime,
    required this.chargingTime,
  });

  factory NoChargingBean.fromJson(Map<String, dynamic> json) {
    return NoChargingBean(
      blackoutTime: json['blackoutTime']?.toString(),
      chargingTime: json['chargingTime']?.toString(),
    );
  }

  Map<String, dynamic> toJson() => {
        'blackoutTime': blackoutTime,
        'chargingTime': chargingTime,
      };
}

@immutable
class RestartTheChargerRegularlyBean {
  final int open;
  final String? time;

  const RestartTheChargerRegularlyBean({
    required this.open,
    required this.time,
  });

  factory RestartTheChargerRegularlyBean.fromJson(Map<String, dynamic> json) {
    return RestartTheChargerRegularlyBean(
      open: (json['open'] as num?)?.toInt() ?? 0,
      time: json['time']?.toString(),
    );
  }

  Map<String, dynamic> toJson() => {
        'open': open,
        'time': time,
      };
}

@immutable
class HeatingBean {
  final int pmsAutoControl;
  final ApkControlBean? apkControl;

  const HeatingBean({
    required this.pmsAutoControl,
    required this.apkControl,
  });

  factory HeatingBean.fromJson(Map<String, dynamic> json) {
    return HeatingBean(
      pmsAutoControl: (json['pmsAutoControl'] as num?)?.toInt() ?? 0,
      apkControl: json['apkControl'] is Map<String, dynamic>
          ? ApkControlBean.fromJson(
              Map<String, dynamic>.from(json['apkControl'] as Map),
            )
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'pmsAutoControl': pmsAutoControl,
        'apkControl': apkControl?.toJson(),
      };
}

@immutable
class ApkControlBean {
  final int open;
  final int permit;
  final int prohibit;

  const ApkControlBean({
    required this.open,
    required this.permit,
    required this.prohibit,
  });

  factory ApkControlBean.fromJson(Map<String, dynamic> json) {
    return ApkControlBean(
      open: (json['open'] as num?)?.toInt() ?? 0,
      permit: (json['permit'] as num?)?.toInt() ?? 0,
      prohibit: (json['prohibit'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'open': open,
        'permit': permit,
        'prohibit': prohibit,
      };
}

@immutable
class HomePageDetailsBean {
  final int show;

  const HomePageDetailsBean({
    required this.show,
  });

  factory HomePageDetailsBean.fromJson(Map<String, dynamic> json) {
    return HomePageDetailsBean(
      show: (json['show'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'show': show,
      };
}

@immutable
class MicroSwitchBean {
  final int open;

  const MicroSwitchBean({
    required this.open,
  });

  factory MicroSwitchBean.fromJson(Map<String, dynamic> json) {
    return MicroSwitchBean(
      open: (json['open'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'open': open,
      };
}

@immutable
class BatteryConnectorBean {
  final int open;

  const BatteryConnectorBean({
    required this.open,
  });

  factory BatteryConnectorBean.fromJson(Map<String, dynamic> json) {
    return BatteryConnectorBean(
      open: (json['open'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'open': open,
      };
}

@immutable
class ChargingStrategyBean {
  final int type;
  final int level;

  const ChargingStrategyBean({
    required this.type,
    required this.level,
  });

  factory ChargingStrategyBean.fromJson(Map<String, dynamic> json) {
    return ChargingStrategyBean(
      type: (json['type'] as num?)?.toInt() ?? 0,
      level: (json['level'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'type': type,
        'level': level,
      };
}

@immutable
class MinimumElectricityBorrowedBean {
  final int level;

  const MinimumElectricityBorrowedBean({
    required this.level,
  });

  factory MinimumElectricityBorrowedBean.fromJson(Map<String, dynamic> json) {
    return MinimumElectricityBorrowedBean(
      level: (json['level'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'level': level,
      };
}

@immutable
class MaximumChargeCapacityBean {
  final int level;

  const MaximumChargeCapacityBean({
    required this.level,
  });

  factory MaximumChargeCapacityBean.fromJson(Map<String, dynamic> json) {
    return MaximumChargeCapacityBean(
      level: (json['level'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'level': level,
      };
}
